src/Integration/LocalServices/ProjectSystemFilter.cs

```diff
@@ -34,48 +34,35 @@ public ProjectSystemFilter(IHost host)
 
         #region IProjectSystemFilter
 
-        public bool IsAccepted(DteProject dteProject)
+        public bool IsAccepted(DteProject project)
         {
-            if (dteProject == null)
+            if (project == null)
             {
-                throw new ArgumentNullException(nameof(dteProject));
+                throw new ArgumentNullException(nameof(project));
             }
 
-            var projectName = dteProject.Name;
-            var hierarchy = this.projectSystem.GetIVsHierarchy(dteProject);
+            var projectName = project.Name;
+            var hierarchy = this.projectSystem.GetIVsHierarchy(project);
             var propertyStorage = hierarchy as IVsBuildPropertyStorage;
 
             if (hierarchy == null || propertyStorage == null)
             {
-                throw new ArgumentException(Strings.ProjectFilterDteProjectFailedToGetIVs, nameof(dteProject));
+                throw new ArgumentException(Strings.ProjectFilterDteProjectFailedToGetIVs, nameof(project));
             }
 
             // Accept only supported languages
-            var language = Language.ForProject(dteProject);
-            if (language == null || !language.IsSupported)
+            if (IsNotSupportedProject(project))
             {
                 return false;
             }
 
-            // General exclusions
-            // If exclusion property is set, this takes precedence
-            bool? sonarExclude = GetPropertyBool(propertyStorage, Constants.SonarQubeExcludeBuildPropertyKey);
-            if (sonarExclude.HasValue)
+            if (IsExcludedViaProjectProperty(propertyStorage))
             {
-                return !sonarExclude.Value;
-            }
-
-            // Ignore test projects
-            // If specifically marked with test project property, use that to specifiy if test project or not
-            bool? sonarTest = GetPropertyBool(propertyStorage, Constants.SonarQubeTestProjectBuildPropertyKey);
-            if (sonarTest.HasValue)
-            {
-                return !sonarTest.Value;
+                return false;
```

Comment:
@duncanpMS, was there a JIRA issue raised about the inconsistencies between the scanner, SonarLint and SonarAnalyzer (in VSIX mode) w.r.t. test detection?